### PR TITLE
feat(TextArea/react): add component

### PR DESCRIPTION
**Pull request by haquezameer**

### CODE REVIEW

File: src/atoms/TextArea/TextArea.web.js [hunk 1]
```diff
@@ -1,3 +1,91 @@
...
+  iconLeft,
+  iconRight,
+  maxLength,
+  label,
+  testID,
+  labelPosition,
+  width,
+}) => {
+  if (!isEmpty(prefix) && !isEmpty(iconLeft)) {
+    throw Error('Cannot have prefix and left icon together');
```
Comment (dhruvdutt) [inline src/atoms/TextArea/TextArea.web.js:26]: Move to propTypes validations.

File: src/atoms/TextInput/TextInput.web.js [hunk 2]
```diff
@@ -284,10 +311,12 @@ const TextInput = ({
     }
   }, [inputRef, containerRef]);
 
+  const _rows = variant === 'outlined' ? 1 : 3;
```
Comment (dhruvdutt) [inline src/atoms/TextInput/TextInput.web.js:314]: What is this constraint? Is it in design?

File: src/atoms/TextInput/TextInput.web.js [hunk 3]
```diff
@@ -244,8 +262,17 @@ const TextInput = ({
...
         onChange(text);
       }
+
+      if (_isMultiline && variant === 'outlined') {
+        setTimeout(() => {
+          inputRef.current.style.height = 'inherit';
+          // Calculate the height
+          const height = inputRef.current.scrollHeight;
+          inputRef.current.style.height = `${height}px`;
+        }, 0);
```
Comment (dhruvdutt) [inline src/atoms/TextInput/TextInput.web.js:272]: Why is `setTimeout` needed here?

File: src/atoms/TextInput/TextInput.web.js [hunk 4]
```diff
@@ -244,8 +262,17 @@ const TextInput = ({
       if (onChange) {
         onChange(text);
       }
+
+      if (_isMultiline && variant === 'outlined') {
+        setTimeout(() => {
+          inputRef.current.style.height = 'inherit';
+          // Calculate the height
+          const height = inputRef.current.scrollHeight;
+          inputRef.current.style.height = `${height}px`;
```
Comment (dhruvdutt) [inline src/atoms/TextInput/TextInput.web.js:271]: Not able to understand what this whole block of code is doing. Please add a comment.
Also, we're setting `inputRef.current.style.height` twice? First `inherit` and then `${height}px`. Why?

File: src/atoms/TextArea/TextArea.web.js [hunk 5]
```diff
@@ -1,3 +1,91 @@
...
+      suffix={suffix}
+      disabled={disabled}
+      iconLeft={iconLeft}
+      iconRight={iconRight}
+      maxLength={maxLength}
+      label={label}
+      labelPosition={labelPosition}
+      width={width}
+      value={value}
+      _isMultiline={true}
```
Comment (dhruvdutt) [inline src/atoms/TextArea/TextArea.web.js:55]:
```suggestion
      _isMultiline
```

File: src/atoms/TextInput/TextInput.web.js [hunk 6]
```diff
@@ -50,6 +50,21 @@ const styles = {
       }
       return '';
     },
+    height({ _isMultiline, variant }) {
+      if (variant === 'filled') {
+        if (_isMultiline) {
+          return '64px';
```
Comment (dhruvdutt) [inline src/atoms/TextInput/TextInput.web.js:56]: Use `makePxValue` function.

File: src/atoms/TextInput/TextInput.web.js [hunk 7]
```diff
@@ -50,6 +50,21 @@ const styles = {
...
+        }
+        return '36px';
```
Comment (dhruvdutt) [inline src/atoms/TextInput/TextInput.web.js:58]: Use `makePxValue` function.

File: src/atoms/TextInput/TextInput.web.js [hunk 8]
```diff
@@ -76,9 +91,11 @@ const StyledInput = styled.input`
   line-height: ${(props) => props.theme.fonts.lineHeight.medium};
   font-family: ${(props) => props.theme.fonts.family.lato.regular};
   color: ${styles.textInput.color};
+  max-height: 64px;
```
Comment (dhruvdutt) [inline src/atoms/TextInput/TextInput.web.js:94]: Is it for 3 rows?

File: src/atoms/TextInput/TextInput.web.js (hunk 3 above)
Comment (haquezameer) [reply src/atoms/TextInput/TextInput.web.js]: removed, not required.

File: src/atoms/TextInput/TextInput.web.js (hunk 4 above)
Comment (haquezameer) [reply src/atoms/TextInput/TextInput.web.js]: have moved this to a useEffect. It adjusts the height of textarea based on the input. 
Have added comments.

File: src/atoms/TextInput/TextInput.web.js (hunk 8 above)
Comment (haquezameer) [reply src/atoms/TextInput/TextInput.web.js:94]: from design max-height is 64px

File: src/atoms/TextInput/TextInput.web.js (hunk 2 above)
Comment (haquezameer) [reply src/atoms/TextInput/TextInput.web.js:314]: had to add _rows=1 for textarea with outlined variant to have its initial size smaller. for variant `filled` 3 rows what comes under max-height of 64px.

File: src/atoms/TextInput/TextInput.web.js [hunk 9]
```diff
@@ -370,7 +398,7 @@ const TextInput = ({
                                   hasText,
                                 })}
                               >
-                                <Size minWidth={[0]}>
+                                <Size minWidth={0} maxHeight={8}>
```
Comment (dhruvdutt) [inline src/atoms/TextInput/TextInput.web.js:401]: Why is `maxHeight` needed?

File: src/atoms/TextArea/__tests__/TextArea.web.test.js [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+    });
+
+    it('renders filled variant with default props', () => {
+      const { container } = renderWithTheme(<TextArea variant="filled" />);
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('id', () => {
+    it('should not have id attribute if not provded as prop', () => {
```
Comment (haquezameer) [inline src/atoms/TextArea/__tests__/TextArea.web.test.js:26]: typo. will fix.

Comment (dhruvdutt) [reply src/atoms/TextArea/__tests__/TextArea.web.test.js:26]: Looks good. Fix the typo and make sure to address comments from text input PR test file changes here.

File: src/atoms/TextArea/__tests__/TextArea.web.test.js [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+    it('should update placeholder if placeholder is passed as prop', () => {
+      const placeholder = 'Example Placeholder';
+      const { getByPlaceholderText } = renderWithTheme(<TextArea placeholder={placeholder} />);
+      const textArea = getByPlaceholderText(placeholder);
+      expect(textArea.placeholder).toEqual(placeholder);
+    });
+  });
+
+  describe('width', () => {
+    it('renders TextArea with medium(default) width', () => {
```
Comment (dhruvdutt) [inline src/atoms/TextArea/__tests__/TextArea.web.test.js:111]: at some places `it('should..')` and some places `it('renders..')`

File: src/atoms/TextArea/__tests__/TextArea.web.test.js [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+      const textArea = getByLabelText(SAMPLE_LABEL);
+      textArea.focus();
+      expect(textArea).toHaveFocus();
+    });
+
+    it('should not have focus after TextArea blur', () => {
+      const { getByLabelText } = renderWithTheme(<TextArea label={SAMPLE_LABEL} id={SAMPLE_ID} />);
+      const textArea = getByLabelText(SAMPLE_LABEL);
+
+      // should have focus initially
```
Comment (dhruvdutt) [inline src/atoms/TextArea/__tests__/TextArea.web.test.js:139]: The comment looks slightly misleading. Focus is not initially. We are setting it. I think comment is not needed. Code is pretty self explanatory.

File: src/atoms/TextArea/__tests__/TextArea.web.test.js [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+    it('should update testID if testID is provided as prop', () => {
+      const testID = 'sample-test-id';
+      const { queryByTestId } = renderWithTheme(<TextArea testID={testID} />);
+      expect(queryByTestId('ds-text-area')).toBeNull();
+      expect(queryByTestId(testID)).not.toBeNull();
+    });
+  });
+
+  describe('error', () => {
+    it('should throw error when labelPosition === left & variant === outlined', () => {
```
Comment (dhruvdutt) [inline src/atoms/TextArea/__tests__/TextArea.web.test.js:256]:
```suggestion
    it('should throw error when labelPosition left and variant outlined', () => {
```

**Pull request merged**